src/sentry/utils/dates.py

```diff
@@ -104,6 +103,37 @@ def parse_timestamp(value: Any) -> datetime | None:
     return rv.replace(tzinfo=UTC)
 
 
+def parse_timestamp(value: datetime | int | float | str | bytes | None) -> datetime | None:
+    # TODO(mitsuhiko): merge this code with coreapis date parser
+    if not value:
+        return None
+    elif isinstance(value, datetime):
+        return value
+    elif isinstance(value, (int, float)):
+        return datetime.fromtimestamp(value, UTC)
+
+    expected_value = _parse_timestamp_old(value)
+
+    # TODO(@anonrig): Remove this once we make sure both results return the same.
+    if options.get("system.use-date-util-timestamps"):
+        try:
+            if isinstance(value, bytes):
+                value = value.decode()
+            new_value = parse(value, ignoretz=True).replace(tzinfo=UTC)
+        except (ParserError, ValueError) as err:
+            logger.exception("parse_timestamp", extra={"error": str(err)})
```

Comment:
Nit: `exception` will already capture all the exception details, so no need for the extra here

Reply:
nice